### PR TITLE
Add initial ability to generate metadata source from a symbol

### PR DESCRIPTION
**Pull request by Andorbal**
This is using functionality from the Roslyn.Features assembly that aren't public

### CODE REVIEW

File: src/OmniSharp/Api/v1/Navigation/OmnisharpController.GotoDefinition.cs [hunk 1]
```diff
@@ -27,17 +31,54 @@ public partial class OmnisharpController
...
                     {
-                        FileName = lineSpan.Path,
-                        Line = lineSpan.StartLinePosition.Line + 1,
-                        Column = lineSpan.StartLinePosition.Character + 1
-                    };
+                        var temporaryDocument = document.Project.AddDocument("foo", string.Empty);
+                        var topLevelSymbol = GetTopLevelContainingNamedType(symbol);
+
+                        var assembly = Assembly.Load("Microsoft.CodeAnalysis.CSharp.Features, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35");
+                        var type = assembly.GetType("Microsoft.CodeAnalysis.CSharp.MetadataAsSource.CSharpMetadataAsSourceService");
```
Comment (nosami) [inline src/OmniSharp/Api/v1/Navigation/OmnisharpController.GotoDefinition.cs]: we should probably cache the type

**Comment by david-driscoll** [comment]
I love this... I have plans for how to use this in atom too... we will see! :dart:

**Comment by nosami** [comment]
Looks Travis has found a genuine problem - https://travis-ci.org/OmniSharp/omnisharp-roslyn/builds/69991483#L3404

**Comment by Andorbal** [comment]
The test failed because I was being too specific; apparently, the mono people list the List<T>'s interfaces in a different order than Microsoft does.  While I was at it, I extracted the assembly loading code into a Lazy<T> static field.

**Comment by nosami** [comment]
closed in favour of #263

**Pull request closed without merging**